cmake/modules/CompileWarnings.cmake

```diff
@@ -1,9 +1,17 @@
 macro(wpilib_target_warnings target)
     if(NOT MSVC)
-        target_compile_options(
-            ${target}
-            PRIVATE -Wall -pedantic -Wextra -Werror -Wno-unused-parameter ${WPILIB_TARGET_WARNINGS}
+        set(WARNING_FLAGS
+            -Wall
+            -pedantic
+            -Wextra
+            -Wno-unused-parameter
+            ${WPILIB_TARGET_WARNINGS}
         )
+        if(NOT NO_WERROR)
+            set(WARNING_FLAGS "${WARNING_FLAGS}" -Werror)
```

Comment:
Is it correct to quote this?

Reply:
It might not be, I forget how CMake handles this. It builds properly, but I'm not sure it's getting all the flags into the compiler; I'll test it out

Reply:
Regardless of whether or not there are quotes, compilation flags are handled properly with or without NO_WERROR. I believe it matters only when you add the compile options

Reply:
Quotations here seem incorrect because it'll expand to one string instead of a list.

https://stackoverflow.com/a/35853080

Reply:
It technically works but yeah I'll correct it anyways.